diff --git a/index.js b/index.js
index 3164b1d0b40f7873aadb91798471a35f0d422405..4e8a914473d9176589b25add0c6fdebc0f30324a 100755
--- a/index.js
+++ b/index.js
@@ -7,7 +7,7 @@ const findCacheDir = require('find-cache-dir')
 const fs = require('./lib/fs-promises')
 const os = require('os')
 const { debuglog, promisify } = require('util')
-const glob = promisify(require('glob'))
+const glob = require('glob')
 const Hash = require('./lib/hash')
 const libCoverage = require('istanbul-lib-coverage')
 const libHook = require('istanbul-lib-hook')
diff --git a/package.json b/package.json
index 31478e7591c1c878ccf638dc13d9cd126f7964b0..bdfbee2d6d5a0e80b88ddc449c0cbb8edfda656d 100644
--- a/package.json
+++ b/package.json
@@ -65,7 +65,7 @@
     "find-up": "^4.1.0",
     "foreground-child": "^2.0.0",
     "get-package-type": "^0.1.0",
-    "glob": "^7.1.6",
+    "glob": "^9.0.0",
     "istanbul-lib-coverage": "^3.0.0",
     "istanbul-lib-hook": "^3.0.0",
     "istanbul-lib-instrument": "^4.0.0",
